### PR TITLE
OY2-18870: Replicate Health Home Measures for FFY 2022

### PR DESCRIPTION
**Pull request by karla-vm**
## Purpose

The 2021 Health Home measures are being replicated and made available for FFY 2022. Users should be able to go in and report on 2022 HH core sets.

Application endpoint:  https://dgmeuzoy53oeq.cloudfront.net/

#### Linked Issues to Close

Closes [OY2-18870](https://qmacbis.atlassian.net/browse/OY2-18870).

## Approach

Replicating the 2021 Health Home measures for FFY 2022 allows users to report on that fiscal year.

## Learning

For this change, I paired with @abelzer212 to understand the changes required from 2021 to 2022.

## Assorted Notes/Considerations

N/A

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/measures/2022/CommonQuestions/MeasurementSpecification/index.tsx [hunk 1]
```diff
@@ -19,10 +19,6 @@ const HEDISChildren = () => {
         label="Specify the version of HEDIS measurement year used:"
         placeholder="Select option"
         options={[
-          {
-            displayValue: "HEDIS MY 2021 (FFY 2022 Core Set Reporting)",
-            value: DC.HEDIS_MY_2021,
-          },
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/MeasurementSpecification/index.tsx:25]: This needs to be left in

File: services/ui-src/src/measures/2022/CommonQuestions/MeasurementSpecification/index.tsx [hunk 2]
```diff
@@ -31,6 +27,10 @@ const HEDISChildren = () => {
             displayValue: "HEDIS 2020 (FFY 2020 Core Set Reporting)",
             value: DC.HEDIS_2020,
           },
+          {
+            displayValue: "HEDIS 2019 (FFY 2019 Core Set Reporting)",
+            value: DC.HEDIS_2019,
+          },
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/MeasurementSpecification/index.tsx:33]: This needs to be removed

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 3]
```diff
@@ -38,7 +38,6 @@ const TotalNDR = ({
     customDenominatorLabel,
     customNumeratorLabel,
     customRateLabel,
-    rateCalculation,
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:41]: I believe this needs to be left in

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 4]
```diff
@@ -74,7 +82,6 @@ const TotalNDR = ({
         customNumeratorLabel={customNumeratorLabel}
         customDenominatorLabel={customDenominatorLabel}
         customRateLabel={customRateLabel}
-        rateCalc={rateCalculation}
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:77]: Same with this one

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 5]
```diff
@@ -141,7 +148,6 @@ const useStandardRateArray: RateArrayBuilder = (name) => {
     customMask,
     performanceMeasureArray,
     IUHHPerformanceMeasureArray,
-    rateCalculation,
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:144]: This also needs to stay

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 6]
```diff
@@ -199,7 +211,6 @@ const useStandardRateArray: RateArrayBuilder = (name) => {
               customDenominatorLabel={customDenominatorLabel}
               customRateLabel={customRateLabel}
               customMask={customMask}
-              rateCalc={rateCalculation}
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:202]: This needs to stay too

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 7]
```diff
@@ -224,9 +234,9 @@ const useQualRateArray: RateArrayBuilder = (name) => {
     calcTotal,
     allowNumeratorGreaterThanDenominator,
     customMask,
-    rateCalculation,
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:227]: Also needs to stay

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 8]
```diff
@@ -254,10 +264,35 @@ const useQualRateArray: RateArrayBuilder = (name) => {
           customDenominatorLabel={customDenominatorLabel}
           customRateLabel={customRateLabel}
           customMask={customMask}
-          rateCalc={rateCalculation}
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:257]: This needs to stay too

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 9]
```diff
@@ -293,9 +327,8 @@ const useAgeGroupsCheckboxes: CheckBoxBuilder = (name) => {
         displayValue: value,
         children: [
           <CUI.Heading key={`${name}.rates.${cleanedLabel}Header`} size={"sm"}>
-            {customPrompt ??
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:296]: This also needs to stay, and the revision needs to not be there

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 10]
```diff
@@ -271,8 +306,7 @@ const useQualRateArray: RateArrayBuilder = (name) => {
  */
 const useAgeGroupsCheckboxes: CheckBoxBuilder = (name) => {
   const options: QMR.CheckboxOption[] = [];
-  const { categories, qualifiers, calcTotal, customPrompt } =
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:274]: Custom prompt needs to stay here

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 11]
```diff
@@ -404,11 +459,9 @@ const useRenderOPMCheckboxOptions = (name: string) => {
     rateMultiplicationValue,
     customMask,
     allowNumeratorGreaterThanDenominator,
-    rateCalculation,
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:407]: Rate Calculation and custom prompt should stay

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 12]
```diff
@@ -451,9 +483,8 @@ const useRenderOPMCheckboxOptions = (name: string) => {
             key={`${name}.rates.${cleanedFieldName}Header`}
             size={"sm"}
           >
-            {customPrompt ??
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:454]: Custom Prompt needs to stay

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 13]
```diff
@@ -421,27 +474,6 @@ const useRenderOPMCheckboxOptions = (name: string) => {
   OPM?.forEach(({ description }, idx) => {
     if (description) {
       const cleanedFieldName = cleanString(description);
-      const RateComponent = (
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:424]: I don't think this should be removed but check with @daniel-belcher on that

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 14]
```diff
@@ -463,7 +494,24 @@ auto-calculate:`}
           >
             Please review the auto-calculated rate and revise if needed.
           </CUI.Heading>,
-          RateComponent,
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:466]: This is where the custom rate component is brought in

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 15]
```diff
@@ -500,14 +548,14 @@ export const NDRSets = ({ name }: NdrProps) => {
         children.push(<AgeGroupNDRSets name={name} key={name} />);
       }
       break;
-    case "AAB":
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:503]: The AAB Case should stay I believe

**Review by abelzer212 — CHANGES_REQUESTED** [review]

File: services/ui-src/src/measures/2022/CommonQuestions/types.ts [hunk 16]
```diff
@@ -15,10 +15,7 @@ export interface MeasurementSpecification {
     | typeof DC.HRSA
     | typeof DC.PQA;
   [DC.MEASUREMENT_SPECIFICATION_HEDIS]: // if Measure Spec is NCQA/HEDIS -> which version are they using
-  | typeof DC.HEDIS_MY_2021
```
Comment (abelzer212) [inline services/ui-src/src/measures/2022/CommonQuestions/types.ts:18]: This should stay and the revision should not be done

File: services/ui-src/src/measures/2022/globalValidations/validateNoNonZeroNumOrDenom/index.ts [hunk 17]
```diff
@@ -8,13 +8,14 @@ import {
...
   getOtherPerformanceMeasureRateArray,
+  // getPerfMeasureRateArray,
 } from "../dataDrivenTools";
 
 interface ValProps extends UVFP {
   hybridData?: boolean;
 }
 
-const _validationRateNotZero = ({ location, rateData }: UVFP) => {
+export const _validationRateNotZero = ({ location, rateData }: UVFP) => {
```
Comment (jphechter) [inline services/ui-src/src/measures/2022/globalValidations/validateNoNonZeroNumOrDenom/index.ts:18]: Is there a reason we're exporting this? The design pattern in our globalValidation functions is to have private `_validation` functions that do the actual validation check. They should be invoked by a specific PM or OMS function. This function `_validationRateNotZero` should never be called outside this file.
```suggestion
const _validationRateNotZero = ({ location, rateData }: UVFP) => {
```

File: services/ui-src/src/measures/2022/globalValidations/validateNoNonZeroNumOrDenom/index.ts [hunk 18]
```diff
@@ -38,7 +39,11 @@ const _validationRateNotZero = ({ location, rateData }: UVFP) => {
   return errorArray;
 };
 
-const _validationRateZero = ({ location, rateData, hybridData }: ValProps) => {
+export const _validationRateZero = ({
+  location,
+  rateData,
+  hybridData,
+}: ValProps) => {
```
Comment (jphechter) [inline services/ui-src/src/measures/2022/globalValidations/validateNoNonZeroNumOrDenom/index.ts:46]: Same as above.
```suggestion
const _validationRateZero = ({
  location,
  rateData,
  hybridData,
}: ValProps) => {
```

**Review by jphechter — CHANGES_REQUESTED** [review]
I think there are some changes in here that we don't actually want to include. Let me know if we need to talk through anything.

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx (hunk 15 above)
Comment (jphechter) [reply services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:503]: AAB-AD does not have it's own componentFlag, so this should actually be safe to remove.

**Review by abelzer212 — APPROVED** [review]

File: services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 19]
```diff
@@ -453,7 +516,7 @@ const useRenderOPMCheckboxOptions = (name: string) => {
           >
             {customPrompt ??
               `Enter a number for the numerator and the denominator. Rate will
-auto-calculate:`}
+            auto-calculate`}
```
Comment (jphechter) [inline services/ui-src/src/measures/2022/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:519]: Total nit, but do we need to keep this colon?
```suggestion
            auto-calculate:`}
```

**Review by jphechter — APPROVED** [review]

**Comment by benjaminpaige** [comment]
just the failing cypress tests to resolve. stemming from this assertion failure `AssertionError: Timed out retrying after 15000ms: expected '<button.chakra-button.css-1yhq6yr>' to be 'disabled'`

prob a classname issue

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**